### ISSUE TITLE
inject failure through HTTP

### ISSUE THREAD

**Issue by siddontang** [tikv/tikv#4388]
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now we have already installed lots of fail points and used them in tests, but only in uint or local integrated tests, we can't use them in Schrodinger or at runtime. 

**Describe the feature you'd like:**

We can build a release version with fail support, add HTTP handlers to help us trigger/list/delete fail points.   

Refer https://github.com/etcd-io/gofail#http-endpoint

**Comment by kennytm** [comment]
Seems this issue is more suitable for https://github.com/pingcap/fail-rs?

> TODO
>
> Triggering a fail point via the HTTP API is planned but not implemented yet.

**Comment by siddontang** [comment]
@kennytm 

I don't like the way of using an HTTP environment to start a fail service. 

Another reason is that I don't like using another port for `fail` separately, we should use the same HTTP port of TiKV, but unlike Go, there is no standard HTTP lib so we have to write our HTTP logic directly in TiKV.

**Comment by liufuyang** [comment]
Is this issue easy to solve? Can I try give some help? :)

**Comment by BusyJay** [comment]
Yes, it's quite easy. What needs to do is to setup the failpoint registry when TiKV starts, and then provide HTTP interface at status server.

https://github.com/tikv/tikv/blob/3f10327e648b4c1c6935154f8bca4e2abd894419/src/server/status_server.rs#L65-L73

**Comment by liufuyang** [comment]
Okay I seem to have some time to look at this issue now.
The question I have is, how would a user uses the http failpoint when finished? So one can use it like this?:
```
curl SERVER_IP:PORT/fail/trigger -XPUT -d'{"name": "", "actions": "some actions"}' 
```

**Comment by BusyJay** [comment]
How about put the name in the URL part?

**Comment by liufuyang** [comment]
If that is easier when using, I can try that as well :)

**Comment by liufuyang** [comment]
So I made some demo http code here https://github.com/liufuyang/fail-rs-http-demo
Do you mind go take a look at it to see whether that is the way we would like to implement here for tikv?
